Frameworks/CoreGraphics/CGContext.mm

```diff
@@ -2032,15 +2032,76 @@ void CGContextDrawTiledImage(CGContextRef context, CGRect rect, CGImageRef image
 #pragma endregion
 
 #pragma region Drawing Operations - Gradient + Shading
+/*
+* Convert CGGradient to D2D1_GRADIENT_STOP
+*/
+static inline std::vector<D2D1_GRADIENT_STOP> __CGGradientToD2D1GradientStop(CGGradientRef gradient, CGGradientDrawingOptions options) {
+    unsigned long gradientCount = _CGGradientGetCount(gradient);
+
+    std::vector<D2D1_GRADIENT_STOP> gradientStops((options == kCGGradientDrawsAfterEndLocation) ? gradientCount + 1 : gradientCount);
+
+    float* colorComponenets = _CGGradientGetColorComponents(gradient);
+    float* locations = _CGGradientGetStopLocation(gradient);
+    for (unsigned long i = 0; i < gradientCount; ++i) {
+        unsigned int colorIndex = (i * 4);
+        gradientStops[i].color = D2D1::ColorF(colorComponenets[colorIndex],
+                                              colorComponenets[colorIndex + 1],
+                                              colorComponenets[colorIndex + 2],
+                                              colorComponenets[colorIndex + 3]);
+        gradientStops[i].position = locations[i];
+    }
+
+    if (options == kCGGradientDrawsAfterEndLocation) {
+        // Set the edge to be just a bit lower on the location
+        gradientStops[0].position = 1E-45;
```

Comment:
sadly D2D is interpreting that as 0

---
In reply to: [92517563](https://github.com/Microsoft/WinObjC/pull/1535#discussion_r92517563) [](ancestors = 92517563)